src/ImmersedBoundaries/grid_fitted_bottom.jl

```diff
@@ -17,46 +17,43 @@ abstract type AbstractGridFittedBottom{H} <: AbstractGridFittedBoundary end
 
 # To enable comparison with PartialCellBottom in the limiting case that
 # fractional cell height is 1.0.
-struct CenterImmersedCondition end
 struct InterfaceImmersedCondition end
-
-Base.summary(::CenterImmersedCondition) = "CenterImmersedCondition"
-Base.summary(::InterfaceImmersedCondition) = "InterfaceImmersedCondition"
+struct CenterImmersedCondition end
 
 struct GridFittedBottom{H, I} <: AbstractGridFittedBottom{H}
-    bottom_height :: H
+    z_bottom :: H
```

Comment:
I though we want to refer to `z` as the coordinate and height as a positive value. 
At least, this is the direction we are taking in ClimaOcean where we are distinguishing very carefully between `z` and `h` or `height`. 
In this case, this is technically a coordinate, not a height, for example we have now
https://github.com/CliMA/Oceananigans.jl/blob/989108a6205bd39f7b215b2145ff5111015610df/src/ImmersedBoundaries/grid_fitted_bottom.jl#L109 which conflates the meaning of z with height. I thought we had decided this over at ClimaOcean and I wanted to update this code to reflect the decision we had made.

Reply:
> I though we want to refer to z as the coordinate and height as a positive value.

Usually "z" is called "geopotential height". What would you call "z" instead? Altitude?

Reply:
> this is technically a coordinate, not a height

Another name for z is "height coordinate".

Reply:
As someone who hasn't used immersed boundaries that much, I always find it a bit confusing that it's called bottom height because it makes me think I'm meant to put in a positive "height" from the bottom of the domain, but I would know that "z" is meant to be from the top